packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/SurroundingPairScopeHandler/SurroundingPairInteriorScopeHandler.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Direction,
+  Position,
+  TextEditor,
+  type SurroundingPairInteriorScopeType,
+} from "@cursorless/common";
+import { TokenTarget } from "../../../targets";
+import { BaseScopeHandler } from "../BaseScopeHandler";
+import { TargetScope } from "../scope.types";
+import {
+  ScopeIteratorRequirements,
+  type ScopeHandler,
+} from "../scopeHandler.types";
+import type { ScopeHandlerFactory } from "../ScopeHandlerFactory";
+
+export class SurroundingPairInteriorScopeHandler extends BaseScopeHandler {
+  protected isHierarchical = true;
+  private surroundingPairScopeHandler: ScopeHandler;
+
+  constructor(
+    private scopeHandlerFactory: ScopeHandlerFactory,
+    public readonly scopeType: SurroundingPairInteriorScopeType,
+    private languageId: string,
+  ) {
+    super();
+
+    this.surroundingPairScopeHandler = this.scopeHandlerFactory.create(
+      {
+        type: "surroundingPair",
+        delimiter: this.scopeType.delimiter,
+        requireStrongContainment: true,
+      },
+      this.languageId,
+    )!;
+  }
+
+  get iterationScopeType() {
+    return this.surroundingPairScopeHandler.iterationScopeType;
+  }
+
+  *generateScopeCandidates(
+    editor: TextEditor,
+    position: Position,
+    direction: Direction,
+    hints: ScopeIteratorRequirements,
+  ): Iterable<TargetScope> {
+    const scopes = this.surroundingPairScopeHandler.generateScopes(
+      editor,
+      position,
+      direction,
+      hints,
+    );
+
+    for (const scope of scopes) {
+      const interiorRange = scope.getTargets(false)[0].getInterior()![0]
+        .contentRange;
+      yield {
+        editor,
+        domain: interiorRange,
+        getTargets(isReversed) {
+          return [
+            new TokenTarget({
+              editor,
+              isReversed,
+              contentRange: interiorRange,
+            }),
```

Comment:
That was my original intents, but interior targets expects the full interior range which we don't have here.
We need to apply the `isReversed` argument. Otherwise we could have done that.

Reply:
I went ahead and did it and it seemed to work well. It broke some of the new tests but they're all due to the fact we're using paragraph removal which I don't think is correct anyway